action.yaml

```diff
@@ -14,14 +14,31 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - name: install dependencies
-      run: mix deps.get
+    - run: sudo apt-get --assume-yes update 
+      shell: bash
+    - run: sudo apt-get --assume-yes install perl cpanminus
+      shell: bash
+    - run: perl -v
+      shell: bash
+    - run: cpanm --notest --sudo Dist::Zilla
+      shell: bash
+    - run: dzil --version
+      shell: bash
+    - name: install dependencies required for building
+      run: dzil authordeps --missing | cpanm --sudo --notest
       shell: bash
       working-directory: ${{ inputs.working-directory }}
-    - name: publish package
+    - name: build package
       run: |
-        mix hex.publish --yes
-      env:
-        HEX_API_KEY: ${{ inputs.hex-api-key }}
+        echo Building version $(cat ./VERSION)
+        dzil build
```

Comment:
the combination of `dzil build` and `cpan-upload` does not test the generated "dist" (package). Is that the intent? Or do we want to run some level of CI on the generated package/dist?

Reply:
In all of our other publish actions, we assume the commit has been tested and our job is just to build the artefact(s) and publish them.

e.g. https://github.com/cucumber/action-publish-npm/blob/main/action.yaml#L27

When we *use* the action in a workflow, we may well run the tests first, e.g. https://github.com/cucumber/tag-expressions/blob/main/.github/workflows/release-npm.yml#L19

Reply:
Yes. And I agree with that. The "dzil test" action tests the code outside the comfort zone of its repository, meaning that any packaging errors show up by dzil's testing which may not show up by in-repository testing. I guess that our CI should then both contain "author tests" as well as package tests.